enhancements/baremetal-IPI/baremetal-provisioning-config.md

```diff
@@ -0,0 +1,185 @@
+be within this network. The value for this config is expected to be
+provided in the format : 10.1.0.0/24.
+
+4. ProvisioningDHCPRange : This is a range of IP addresses from which
+baremetal hosts can be assigned their IP address. This range should be
+provided as two comma seperated IP addresses where the addresses represent
+a contiguous range of IPs that are available to be allocated. It is expected
+to be provided in the format : 10.1.0.10, 10.1.0.100.
+
+5. CacheURL : This is a location where image required to bring up baremetal
+hosts have been previously downloaded for faster image downloads within the
+cluster. This is an optional parameter and is expected to be provided
+in the format : http://111.111.111.1/images
+
+6. RHCOSImageURL : This config is expected to be the location of the
+official RHCOS release image. It is a required parameter and will be used
+to deploy new nodes in the cluster.
+
+### User Stories [optional]
+
+This enhancement will allow users to have their Barametal IPI deployments
+customizable to their specific hardware and network requirements.
+
+This also allows IPI baremetal deployments to take place without manual
+workarounds like creating a ConfigMap for the config (which is the
+current approach we are using to test baremetal deployments.) 
+
+The new config items would be set by the installer and will be used by
+the machine-api-operator (MAO) to generate more config items that are
+derivable from these basic parameters. Put together, these config
+parameters are passed in as environment variables to the various containers
+that make up a metal3 baremetal IPI deployment. 
+
+## Design Details
+
+### Test Plan
+
+The test plan should involve making sure the openshift/installer generates
+all configuration items within the BaremetalPlatformStatus when the platform
+type is Baremetal.
+
+MAO reads this configuration and uses these to derive additional configuration
+required to bring up a metal3 cluster. E2e testing should make sure that MAO
+is able to bring up a metal3 cluster using config from this enhanced
+BaremetalPlatformStatus CR.
+
+### Upgrade / Downgrade Strategy
+
+Baremetal Platform type will be available for customers to use for the first
+time in Openshift 4.3. And, when it is installed, it will always start as a
+fresh baremetal installation at least in 4.3. There is no use case where a 4.2
+installation would be upgraded to a 4.3 installation with Baremetal Platform
+support enabled.
+
+For the above reason, in this particular case, we would not have to make
+additional provisions for an upgrade strategy.
+
+Baremetal platform is experimental in 4.2 but, in the small chance that a user
+might upgrade to 4.3 from 4.2 and also set the Platform type to Baremetal for
+for the first time during the upgrade, our upgrade strategy would be to
+support obtaining the config from both the ConfigMap and the new CR.
+In 4.3, the MAO would be implemented to first look for the baremetal config
```

Comment:
Do we need to update this to s/4.3/4.4/g and s/4.2/4.3/g ?

Reply:
The background doc talks about baremetal IPI deployments in tech preview. So, we should say something about 4.2 here I think.

Reply:
My point is it's still in the same state for 4.3, e.g using the configmap unless this work ends up backported.